src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslCachedSystemStoreProvider.cs

```diff
@@ -207,20 +207,17 @@ bool ProcessDir(string dir, out DateTime lastModified)
             bool ProcessFile(string file, out DateTime lastModified, bool skipStat = false)
             {
                 bool readData = false;
-
-                if (skipStat)
-                {
-                    lastModified = default;
-                }
-                else if (!TryStatFile(file, out lastModified))
+                lastModified = default;
+                bool result = TryStatFile(file, out DateTime modifiedDate, out (long, long) fileDetails);
+                if (!skipStat && !result)
```

Comment:
You can remove the `skipStat` argument from this method. We no longer want to ever skip the stat.

And, the changes to this functions can be something like:

```cs
if (!TryStatFile(file, out lastModified, out (long, long) filedId) ||
    !processedFiles.Add(filedId))
{
    return false;
}
```
We don't need to wait till the end of the function to add the file since next time we try to process it, we expect the same result.

Reply:
@tmds I am going to `skiptest` args in next commit. But we can't return `false` if `processedFiles` already have an entry, this will trigger a false positive indicating directory reading is unsuccessful, which will cause to read all certs from `/etc/ssl/certs` (which we don't want as I believe that's the fallback plan).

Rather, I will add the entry to `processedFiles`  if `readdata` is true . Second time, if `processedFiles` has an entry, will return true(basically letting caller know that it was a success) without re-processing it. Hope it clarifies, let me know if you think otherwise.

Reply:
> this will trigger a false positive indicating directory reading is unsuccessful, which will cause to read all certs from /etc/ssl/certs (which we don't want as I believe that's the fallback plan).

That's right, if we return `false`, it messes up `hasStoreData.